pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -475,10 +495,11 @@ def select_accelerator(self) -> Accelerator:
             acc_cls = TPUAccelerator
         else:
             acc_cls = CPUAccelerator
-
+        # as precision_plugin is dependent on training_type_plugin, make sure
+        # that we first select training_type_plugin, then precision_plugin
         return acc_cls(
-            precision_plugin=self.precision_plugin,
             training_type_plugin=self.training_type_plugin,
+            precision_plugin=self.precision_plugin,
         )
```

Comment:
while checking my previous test failure for assert apex precision does not work for sharded plugin (that is actually typo issue), found that this part is fragile also. As when we select precision plugin, it depends on training type plugin (if user do not pass `plugin` in Trainer, instead use `accelerator`), this is problematic, we first need to select the training type plugin. cc @ananthsub , @SeanNare, another motivation for #7324

Reply:
Let's try to move the precision inside before we add the next major plugin.